core/src/window.cpp

```diff
@@ -71,20 +78,58 @@ Window::~Window() {
 }
 
 void Window::window_size_callback(GLFWwindow* window, int width, int height) {
-    Window* self = (Window*)glfwGetWindowUserPointer(window);
+    Window* self = static_cast<Window*>(glfwGetWindowUserPointer(window));
     self->m_width = width;
     self->m_height = height;
     self->recreate_framebuffer(width, height);
     // TODO REPLACE THIS FUNCTION & ALL GLFW CALLBACKS WHEN IN EDITOR
     self->set_window_resize_flag(true);
 }
+
+void Window::window_key_callback(GLFWwindow* window, int key, int scancode, int action, int mods) {
+    Window* self = static_cast<Window*>(glfwGetWindowUserPointer(window));
+    if (action != GLFW_REPEAT)
+        self->m_input.key_event(key, action != GLFW_RELEASE);
+}
+
+void Window::window_char_callback(GLFWwindow* window, unsigned int) {
```

Comment:
Is this finished?

Reply:
some of the stuff in these leo needs to sort out, because he needed the call backs before he could do/finish some of these :)

Reply:
This callback is not for the input manager, but it is needed for when we hook these callbacks also to the editor.